### PR TITLE
chart-testing: init at 3.4.0

### PR DESCRIPTION
**Pull request by AtkinsChang**
[ct](https://github.com/helm/chart-testing) is the the tool for testing Helm charts

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 132904` at 638ec107 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>chart-testing</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `installShellFiles`.
Near pkgs/applications/networking/cluster/helm/chart-testing.nix:1:40:
```
  |
1 | { lib, buildGoModule, fetchFromGitHub, installShellFiles }:
  |                                        ^
```
</li>
  </ul>
</details>

File: pkgs/applications/networking/cluster/helm/chart-testing.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  ldflags = [
+    "-w"
+    "-s"
+    "-X github.com/helm/chart-testing/v3/ct/cmd.Version=${version}"
+    "-X github.com/helm/chart-testing/v3/ct/cmd.GitCommit=${src.rev}"
+    "-X github.com/helm/chart-testing/v3/ct/cmd.BuildDate=19700101-00:00:00"
+  ];
+
+  postInstall = ''
+    mkdir -p $out/etc/ct
```
Comment (legendofmiracles) [inline pkgs/applications/networking/cluster/helm/chart-testing.nix:30]: Not needed with install -D

File: pkgs/applications/networking/cluster/helm/chart-testing.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "chart-testing";
+  version = "v3.4.0";
+
+  src = fetchFromGitHub {
+    owner = "helm";
+    repo = pname;
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/helm/chart-testing.nix:10]:
```suggestion
  version = "3.4.0";

  src = fetchFromGitHub {
    owner = "helm";
    repo = pname;
    rev = "v${version}";
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -25503,6 +25503,8 @@ in
 
   kubernetes-helmPlugins = dontRecurseIntoAttrs (callPackage ../applications/networking/cluster/helm/plugins { });
 
+  chart-testing = callPackage ../applications/networking/cluster/helm/chart-testing.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:25506]:
```suggestion
  chart-testing = callPackage ../applications/networking/cluster/helm/chart-testing { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (AtkinsChang) [reply pkgs/top-level/all-packages.nix:25506]: it is a `.nix` file not directory with `default.nix`

Comment (legendofmiracles) [reply pkgs/top-level/all-packages.nix:25506]: That's the issue

**Pull request merged**